Copyedit documentation

alexshpilkin opened the pull request:
Applies a bit of copyediting polish to the documentation. I think you might want to pull everything in (obviously), but the branch should be cherry-pickable.

There are more typos and things to edit, but I’ll limit myself to this for now. One question: what should be done with the inconsistent usage of focused/focussed? (This PR leaves things as they are.)

dcbaker commented on alot/commands/globals.py:
```diff
@@ -40,14 +40,14 @@
-    prompting to close without sending, and is used by the BufferCloseCommand
-    if settings change after yielding to the UI.
-    """
+    """shut down cleanly"""
```
Please retain the capital `S` and `.` at the end of the first line of the docstring, we try to follow PEP 257 at least somewhat to be consistent with other python projects.

alexshpilkin replied:
Hm. This is not followed for almost all of the other commands, but OK, I’ll put the non-capitalized version (required for consistency of command descriptions) into the decorator instead.

dcbaker replied:
Our documentation (as you noticed) is kinda all over the place.

alexshpilkin replied:
By the way, is there any reason for the occasional difference between between the docstring and the `help=` argument of the decorator? (See `MoveCommand`, `ReloadCommand` in the same file.) If not, perhaps it could be fixed as well.

dcbaker commented on alot/commands/globals.py:
```diff
@@ -40,14 +40,14 @@
+        :param _prompt: For internal use only, used to control prompting to
+        close without sending, and is used by the BufferCloseCommand if
+        settings change after yielding to the UI.
+        """
```
Generally in sphinx documentation the `__init__()` method is document as part of the class documentation, and not separately. It appears that sphinx simply appends the `__init__()` documentation to the class documentation. I don't have a strong opinion either way, as long as we're consistent. @pazz, @lucc opinions?

alexshpilkin replied:
Oh. Missed this, and didn’t know the convention. The change is in any case unnecessary now, because I only moved the parameter description so that it doesn’t leak into the user documentation, but we’ve since agreed to move the user documentation out of the docstring and into the `@registerCommand` decorator.

alexshpilkin commented:
What about the docstring thing in Travis? I have to admit I’ve no idea what it’s complaining about.

dcbaker commented on alot/commands/search.py:
```diff
@@ -135,10 +135,9 @@ def apply(self, ui):
-         'In that case this command will remove the tag from every message in '
-         'the thread.')
+    help='flip presence of tags on this thread: a tag is considered present '
+         'and removed if at least one message in this thread is tagged with '
```
you're missing `will be` between `and` and `removed`, or some other change to make this sentence grammatically correct.

alexshpilkin replied:
I don’t believe I am, unless you insist on the future tense: “a tag (is (considered present) and (removed)) if ...”.

dcbaker replied:
I asked a co-worker to look at this to make sure I'm not way off, but both of us parsed this as "flip present of tags on this thread: a tag is considered (present and removed) if at least one...". We both agree that the original message reads more naturally.

alexshpilkin replied:
”Will be removed“ it is then. I do tend to overcomplicate my sentences sometimes. BTW, the reason for changing it was to turn the whole thing into a single sentence (as it must be if we’re not capitalizing it).

alexshpilkin commented:
Ah, I see what’s wrong. Hang on.

dcbaker commented on docs/source/generate_commands.py:
```diff
@@ -110,7 +110,7 @@ def get_mode_docs():
                            '\n\n' % mode)
         else:
-            header = 'Global Commands'
+            header = 'Global commands'
```
This is a title, please leave `commands` capitalized.

alexshpilkin replied:
None of the sibling titles are capitalized (and I see no way to change this without making the whole thing ugly):

- 2.4 Commands
    - 2.4.1 Global Commands
    - 2.4.2 Commands in _bufferlist_ mode
    - 2.4.3 Commands in _envelope_ mode
    - 2.4.4 Commands in _namedqueries_ mode
    - _&c._

Perhaps I should fix the (non-)capitalization of 2.1, though.

dcbaker replied:
Okay, it's annoying that the TOC renders it that way, because it does look somewhat unnatural there.

alexshpilkin replied:
I actually feel all right without titlecase for tertiary headings. My native language doesn’t use titlecase at all, though.

What do you think should be done with the tertiary titles in section 3? Currently about a half of them are titlecased.

alexshpilkin commented:
Ouch. That is one confusing error message. (Solution: rebase on top of master. The check is failing on the automatic merge commit, not on my branch.)

alexshpilkin commented:
Changed things as discussed, but left “Global commands” non-titlecased. Added a colon in “The following commands are available ... :”. Titlecased 2.1 “Command-Line Invocation” and 3.1 “Configuration Options”.

dcbaker commented:
Merged. Thank you!
